### ISSUE TITLE
use Release.Name in resources

### ISSUE THREAD

**Issue by bluebrown** [elastic/helm-charts#1644]
Hi, it is common practice to use the release name in the created resources, unless the release name is the chart name. I can see that this is not the case with this chart.

This makes it inconvenient to deploy it under a different release name or as a subchart. 

I think it would be good to follow the established pattern. 

Usually this is done by a helper such as https://github.com/elastic/helm-charts/blob/2f4dd45a80e8f3d89aeb66b79edcafb14e090446/elasticsearch/templates/_helpers.tpl#L13

I can see that it's rather using different helpers. For the statefulset it's this for example https://github.com/elastic/helm-charts/blob/2f4dd45a80e8f3d89aeb66b79edcafb14e090446/elasticsearch/templates/_helpers.tpl#L18

**Comment by jslay-excella** [comment]
I agree, this is making deploying it as a child chart a bit problematic, when having to deploy multiple parent charts in the same namespace. 

ergo, `APP` helm chart uses `elasticsearch` helm chart as a dependency. `APP` is *required* to run in one specific namespace. There needs to be two independent deployments if `APP`, `APP1`, and `APP2`.

There is no way to dynamically deploy the two instances without having to *always* override `elasticsearch.cluseterName` or similar. If the cluster name was being derived from `Release.Name`, this would not cause a conflict.

**Comment by jslay-excella** [comment]
bump
